sympy/printing/latex.py

```diff
@@ -1879,15 +1880,15 @@ def _print_NDimArray(self, expr):
         block_str = r'\begin{%MATSTR%}%s\end{%MATSTR%}'
         block_str = block_str.replace('%MATSTR%', mat_str)
         if self._settings['mat_delim']:
-            left_delim = self._settings['mat_delim']
+            left_delim: str = self._settings['mat_delim']
             right_delim = self._delim_dict[left_delim]
             block_str = r'\left' + left_delim + block_str + \
                         r'\right' + right_delim
 
         if expr.rank() == 0:
             return block_str % ""
 
-        level_str = [[]] + [[] for i in range(expr.rank())]
+        level_str: list[list[str]] = [[] for i in range(expr.rank() + 1)]
```

Comment:
minor optimization.
Even if I write `level_str: list[list[str]] = [[]] + [[] for i in range(expr.rank())]`, pyright still thinks `level_str` is `list[list[Unknown]]`.